quickwit-metastore/src/metastore/single_file_metastore.rs

```diff
@@ -51,10 +51,25 @@ fn is_disjoint(left: &Range<i64>, right: &RangeInclusive<i64>) -> bool {
     left.end <= *right.start() || *right.end() < left.start
 }
 
+/// The underlying single file metastore implementation.
+struct InnerSingleFileMetastore {
+    storage: Arc<dyn Storage>,
+    cache: HashMap<String, MetadataSet>,
+}
+
+impl InnerSingleFileMetastore {
+    fn new(storage: Arc<dyn Storage>) -> Self {
+        InnerSingleFileMetastore {
+            storage,
+            cache: HashMap::new(),
+        }
+    }
+}
+
 /// Single file metastore implementation.
 pub struct SingleFileMetastore {
-    storage: Arc<dyn Storage>,
-    cache: Arc<RwLock<HashMap<String, MetadataSet>>>,
+    uri: String,
```

Comment:
the interface of `uri()` is not async and I needed to cache the URI value otherwise there I would have to deal with `await` during locking the `inner`